core/domain/exp_domain.py

```diff
@@ -1342,6 +1342,24 @@ def validate(self, exp_param_specs_dict, allow_null_interaction):
         elif self.interaction.id is not None:
             self.interaction.validate(exp_param_specs_dict)
 
+    def get_training_data(self):
```

Comment:
Here's how to do it: create an exploration (or load a demo one that has training data) and check that get_training_data() is exactly equal to an expected value.